source/android/Renaissance/lib/create_path.dart

```diff
@@ -23,31 +26,99 @@ class _CreatePathPageState extends State<CreatePathPage> {
   var _image;
   bool _isLoading = false;
 
+  var _topics = <Tag>[];
+
   List<TextEditingController> _titleControllers = [];
   List<TextField> _titleFields = [];
   List<TextEditingController> _descControllers = [];
   List<TextField> _descFields = [];
 
+  List<String> _typeOfController = [];
+
   TextEditingController titleController = TextEditingController();
   TextEditingController descriptionController = TextEditingController();
   TextEditingController topicController = TextEditingController();
 
-  void clearMilestones() {
+  void clearItems() {
     _titleControllers = [];
     _titleFields = [];
     _descControllers = [];
     _descFields = [];
+    _typeOfController = [];
+  }
+
+  int findIndex(int i) {
+    String type = _typeOfController[i];
+    int index = 1;
+    for (int ind = 0; ind < i; ind++) {
+      if (_typeOfController[ind] == type) {
+        index += 1;
+      }
+    }
+    return index;
   }
 
-  void _deleteMilestone(int i) {
+  void _deleteItem(int i) {
     setState(() {
       _titleControllers.removeAt(i);
       _titleFields.removeAt(i);
       _descControllers.removeAt(i);
       _descFields.removeAt(i);
+      _typeOfController.removeAt(i);
     });
   }
 
+  void _searchTopic() async {
+    List<Tag> resultTags =
+        await HttpService.shared.searchTopic(topicController.text.trim());
+
+    topicController.clear();
+    return showDialog(
```

Comment:
not a good practice to return something in a void function.

Reply:
Thanks, I should have missed that. Now the function has a proper return type.